Iterate on swe job description (#49)
sourcegraph/careers: job-descriptions/software-engineer.md

### About us

> We're building the new standard developer platform. Top tech companies have invested \$100Ms to build internal developer platforms for code search, code review, alerts, and automation. Sourcegraph provides this standard developer platform to every company, helping startups and large enterprises ship better software faster.

Reviewer: I think reducing the number of times "developer platform" is written can make it better.

```suggestion
Top tech companies like Google and Facebook have invested \$100Ms to build internal developer platforms for code search, code review, alerts, and automation. Sourcegraph is bringing these capabilities to every company in the world, helping both startups and large enterprises ship better software faster.
```

Reviewer 2: Chiming in here to add my 2 cents: I do like the previous opening sentence because it's bold, to the point and sets the tone.

Reviewer: To be clear: I don't feel strong about this. It just sounded a bit alliterative.

Reviewer 3: "$100Ms" feels somewhat strange to me, it's like using casual words in a formal writing. But keep in mind I'm not a native speaker.

Reviewer 3: What happened to our "master plan" and "bring the future sooner"? Is it intentional we're removing these words from this "About us" page?

Author: > I do like the previous opening sentence because it's bold, to the point and sets the tone.

Yeah, I think this is a good property.

1. Bold open ended statement about the product we are building: "new standard developer platform". No qualifications or implied limitations. Full stop.
2. Evidence that this product is valuable (top tech companies have standard developer platforms) and a problem statement (existing dev platforms are internal and not accessible).
3. Our solution solves the problem from (2) by being open and for everyone.

One point that isn't  explicit right now is that we aren't _merely_ bringing what these companies have to everyone else, we are actually improving on it. I think the presence of (1) helps mitigate this, but that it is completely lost in your proposed changes @tsenart .

Overall, this is the language we use on our homepage so it has been thought about by Quinn and I am hesitant to change it here.

> "$100Ms" feels somewhat strange to me, it's like using casual words in a formal writing, because I often see "hundreds of millions".

Yeah it is a bit informal but I think that is ok. I also think a big number like this is more noticeable so it provides desirable emphasis (instead of blending in with the rest of the text).

> What happened to our "master plan" and "bring the future sooner"? Is it intentional we're removing these words from this "About us" page?

I linked to the master plan in the next paragraph and will think about if there is a way to specifically inline the words "bring the future sooner".

> [Our mission](https://sourcegraph.com/plan) is to dramatically increase the number of people who can read and write code. By making code more accessible, we will democratize software development and accelerate important innovations of the future (e.g. transportation, health care, energy, AI, communication, space travel).

Reviewer 2: These two paragraphs are really good!

> [We value openness and transparency](https://docs.sourcegraph.com/dev/open_source_open_company). Our code, our product roadmap, and our company processes are public. This transparency helps us rapidly gather feedback from our customers so we can iterate, learn, and deliver the best product.

Reviewer 4: `Our code` -> add link
`our product roadmap` -> add link
`our company process` -> add link

Author: ```suggestion
We value openness and transparency. [Our code](https://github.com/sourcegraph/sourcegraph), [our product roadmap](https://docs.sourcegraph.com/dev/roadmap), and [our company processes](https://docs.sourcegraph.com/dev/open_source_open_company) are public. This transparency helps us rapidly gather feedback from our customers so we can iterate, learn, and deliver the best product.
```

> To build a product that serves the needs of all developers, we are building a diverse remote-first team that is distributed across the world. Sourcegraph is an equal opportunity workplace; we welcome people from all backgrounds and communities.

Reviewer 2: Absolute nitpick: there's two `build` here. I'd suggest changing one of them into something synonymous, like `To create a product [...]` or `we are growing our diverse [...]`.

Author: ```suggestion
To create a product that serves the needs of all developers, we are building a diverse remote-first team that is distributed across the world. Sourcegraph is an equal opportunity workplace; we welcome people from all backgrounds and communities.
```

We use a variety of technologies to help us accomplish our goals:

- Frontend: TypeScript, React, RxJS, SCSS, Browser extension APIs
- Backend: Go, GraphQL, PostgreSQL, Redis
- Deployment and operations: Kubernetes, Docker, Google Cloud Platform, Terraform

Review our [product roadmap](https://docs.google.com/document/d/1cBsE9801DcBF9chZyMnxRdolqM_1c2pPyGQz15QAvYI/edit) to understand how our engineering team is organized and what projects you might be working on if you started today.

#### Responsibilities

You will:

- Collaborate with the product team to define and prioritize the problems that you will be working to solve.
- Write [RFCs](https://docs.sourcegraph.com/dev/rfcs) to communicate your implementation plan and solicit feedback from teammates.
- Write code to achieve the goals of your projects.
- Provide thoughtful and compassionate feedback to your teammates when reviewing their code and designs.

We will encourage and support you to:

- Collaborate directly with customers to collect feature requirements and resolve issues.
- Publish blog posts and give conference talks about your work at Sourcegraph.

#### Qualifications

> - You are skilled enough within your area of expertise to own the development on a non-trivial change during your first month at Sourcegraph.

Reviewer: ```suggestion
- You are skilled enough within your area of expertise to own the development of a non-trivial change during your first month at Sourcegraph.
```

> - You communcate clearly, especially in writing, and work well on a team.

Reviewer 2: ```suggestion
- You communicate clearly, especially in writing, and work well on a team.
```

> - You are passionate about creating high quality software and understand how to make appropriate tradeoffs (e.g. cut scope) to ship quickly when necessary.

Reviewer 5: ```suggestion
- You are passionate about creating high-quality software and understand how to make appropriate tradeoffs (e.g. cut scope) to ship quickly when necessary.
```
